module/VuFind/src/VuFind/RecordTab/HoldingsILS.php

```diff
@@ -100,13 +100,31 @@ public function getDescription()
     public function getUniqueCallNumbers($items)
     {
         $callNos = [];
-        foreach ($items as $item) {
+        foreach ($items as $i => $item) {
             if (isset($item['callnumber']) && strlen($item['callnumber']) > 0) {
-                $callNos[] = $item['callnumber'];
+                $prefix = $item['callnumber_prefix'] ?? '';
+                $callnumber = $item['callnumber'];
+                $callNos[$i]['callnumber'] = $callnumber;
+                $callNos[$i]['display'] = $prefix
+                    ? $prefix . ' ' . $callnumber : $callnumber;
+                $callNos[$i]['prefix'] = $item['callnumber_prefix'] ?? '';
             }
         }
-        sort($callNos);
-        return array_unique($callNos);
+
+        uasort(
+            $callNos,
+            function ($a, $b) {
+                return $a['display'] <=> $b['display'];
+            }
+        );
+
+        $unique = [];
```

Comment:
I wonder if it would be more efficient to deduplicate before you sort, since the sort operation is probably more expensive than the deduplication, especially if there are large numbers of values to compare with one another.